core/web/log_controller.go

```diff
@@ -0,0 +1,75 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+	"go.uber.org/zap/zapcore"
+)
+
+// LogController manages the logger config
+type LogController struct {
+	App chainlink.Application
+}
+
+type LogPatchRequest struct {
+	Level      string `json:"level"`
+	SqlEnabled *bool  `json:"sqlEnabled"`
+}
+
+// Patch sets a log level and enables sql logging for the logger
+func (cc *LogController) Patch(c *gin.Context) {
+	request := &LogPatchRequest{}
+	if err := c.ShouldBindJSON(request); err != nil {
+		jsonAPIError(c, http.StatusUnprocessableEntity, err)
+		return
+	}
+
+	if request.Level == "" && request.SqlEnabled == nil {
+		jsonAPIError(c, http.StatusInternalServerError, fmt.Errorf("please set either logLevel or logSql as params in order to set the log level"))
+		return
+	}
+
+	if request.Level != "" {
+		var ll zapcore.Level
+		err := ll.UnmarshalText([]byte(request.Level))
+		if err != nil {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return
+		}
+		cc.App.GetStore().Config.Set("LOG_LEVEL", ll.String())
+		err = cc.App.GetStore().SetConfigStrValue("LogLevel", ll.String())
```

Comment:
This DB write should be accepting `c.Request.Context()` as a parameter then used like `orm.DB.WithContext(ctx)` so if they cancel the request the db write gets cancelled too

Reply:
actually furthermore - I think both these DB writes need to be inside a transaction to prevent one succeeding and one failing

Reply:
updated to use gin.Context